File: app/src/main/java/net/wojteksz128/worktimemeasureapp/database/DatabaseConverters.java
package net.wojteksz128.worktimemeasureapp.database;

import android.arch.persistence.room.TypeConverter;

import net.wojteksz128.worktimemeasureapp.database.comeEvent.ComeEventType;

import java.util.Date;

public class DatabaseConverters {

    public static class DateConverter {

        @TypeConverter
        public static Date toDate(Long timestamp) {
            return timestamp == null ? null : new Date(timestamp);
        }

        @TypeConverter
        public static Long toTimestamp(Date date) {
            return date == null ? null : date.getTime();
        }
    }

    public static class ComeEventTypeConverter {

        @TypeConverter
        public static ComeEventType toComeEventType(String type) {
            return type == null ? null : ComeEventType.valueOf(type);
        }

        @TypeConverter
        public static String toString(ComeEventType eventType) {
            return eventType == null ? null : eventType.name();
        }
    }
}

File: app/src/main/kotlin/net/wojteksz128/worktimemeasureapp/database/DatabaseConverters.kt
package net.wojteksz128.worktimemeasureapp.database

import android.arch.persistence.room.TypeConverter
import net.wojteksz128.worktimemeasureapp.database.comeEvent.ComeEventType
import java.util.*

class DatabaseConverters {

    object DateConverter {

        @TypeConverter
        fun toDate(timestamp: Long?): Date? {
            return if (timestamp == null) null else Date(timestamp)
        }

        @TypeConverter
        fun toTimestamp(date: Date?): Long? {
            return date?.time
        }
    }

    object ComeEventTypeConverter {

        @TypeConverter
        fun toComeEventType(type: String?): ComeEventType? {
            return if (type == null) null else ComeEventType.valueOf(type)
        }

        @TypeConverter
        fun toString(eventType: ComeEventType?): String? {
            return eventType?.name
        }
    }
}
